Sparkle/SUAppcast.m

```diff
@@ -229,7 +229,11 @@ - (void)downloadDidFinish:(NSURLDownload *)__unused aDownload
     }
 
     if (failed) {
-        [self reportError:[NSError errorWithDomain:SUSparkleErrorDomain code:SUAppcastParseError userInfo:@{ NSLocalizedDescriptionKey: SULocalizedString(@"An error occurred while parsing the update feed.", nil) }]];
+        [self reportError:[NSError errorWithDomain:SUSparkleErrorDomain
+                                              code:SUAppcastParseError
+                                          userInfo:@{
+                                                     NSLocalizedDescriptionKey: SULocalizedString(@"An error occurred while parsing the update feed.", nil),
+                                                     NSUnderlyingErrorKey: error ? error : [NSNull null]}]];
```

Comment:
This requires careful consideration. It's *very* dangerous to touch an NSError* in the even slightest if an operation succeeded, especially if it's coming from Apple's APIs, and it may not even be nil. Also see Apple's [note](https://developer.apple.com/library/ios/documentation/Cocoa/Conceptual/ErrorHandlingCocoa/CreateCustomizeNSError/CreateCustomizeNSError.html) on the matter.

A boolean should be used to keep track if an error occurred, by only looking at the boolean return values of such operations. I'd also personally avoid reusing the same error pointer for different operations.